### PR TITLE
Display how long the user has been on the wait list

### LINKED ISSUE #131: Check how long I've been on the wait list for a book
**As a** user waiting for a book to become available
**When** I check my wait list
**I want to** see how long I've been on the wait list
**So I can** see if I should talk to the person who has it

Depends on #129.

An option would be to add something like a ribbon on the book component with the relative time the user has been waiting, or the date.

## Acceptance Criteria

**Given** I have books on my wait list
**When** I open the My Books page
**Then** I see how long I've been on the wait list for each book in the Wait list section

### PR DESCRIPTION
**Pull request by macecchi**
Closes #131.

This will display an element on the book component whenever the book is on the current user's wait list.

The API was also modified to return the details of wait list items (containing the date) and the date when the current user has added to the wait list.

image
image

---
This change is [Reviewable](https://reviewable.io/reviews/ayr-ton/kamu/146)

### CODE REVIEW

File: assets/src/components/books/Book.css [hunk 1]
```diff
@@ -78,3 +80,21 @@
...
+    z-index: 2;
+    right: 0;
+    top: 5px;
+    padding: 2px 6px;
+    font-size: 0.8em;
+    border-radius: 0 0 0 4px;
+}
+
+.book .waitlist-indicator .material-icons {
+    font-size: 15px;
```
Comment (lucashanke) [inline assets/src/components/books/Book.css]: Can we make all the font-sizes use the same metric? I think we're using px.

File: assets/src/components/books/Book.css [hunk 2]
```diff
@@ -78,3 +80,21 @@
...
+
+.book .waitlist-indicator {
+    background-color: var(--book-container-background);
+    color: #999;
+    position: absolute;
+    z-index: 2;
+    right: 0;
+    top: 5px;
+    padding: 2px 6px;
+    font-size: 0.8em;
```
Comment (lucashanke) [inline assets/src/components/books/Book.css]: comment is below...

File: assets/src/components/books/Book.test.jsx [hunk 3]
```diff
@@ -180,6 +181,28 @@ describe('Book', () => {
       expect(leaveWaitlist).toHaveBeenCalledWith(book);
       expect(bookComponent).toHaveJoinWaitlistButton();
     });
+
+    it('has a waitlist indicator when book is on users waitlist', async () => {
+      const book = someBookThatIsInMyWaitlist();
+      const bookComponent = createComponent(book);
+
+      const indicator = bookComponent.find(WaitlistIndicator);
```
Comment (lucashanke) [inline assets/src/components/books/Book.test.jsx]: I think we should avoid querying for a component. Let's use test id's instead. That way we don't test the implementation itself.

File: assets/src/components/books/Book.css (hunk 1 above)
Comment (macecchi) [reply assets/src/components/books/Book.css]: All changed to em in 21ece60

File: assets/src/components/books/Book.test.jsx (hunk 3 above)
Comment (macecchi) [reply assets/src/components/books/Book.test.jsx]: adfd63f

**Pull request merged**